src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java

```diff
@@ -33,19 +36,57 @@ public RuleResponse validate(CommunicationMessage message) {
             LocalDateTime currentDateTime = LocalDateTime.now(clock);
 
             BusinessDayConfig businessDayConfig = timeConfigService.getBusinessDayConfig();
-            BusinessTime businessTime = businessDayConfig.getCurrentBusinessTime(currentDateTime);
 
-            LocalTime currentTime = currentDateTime.toLocalTime();
-            LocalTime startTime = businessTime.getStartTime();
-            LocalTime endTime = businessTime.getEndTime();
-            log.debug("start time: {}, end time: {}, current time: {}", startTime, endTime, currentTime);
+            boolean exceptionCharacteristicPresent = false;
+            if (businessDayConfig.getBusinessTime() != null) {
+                exceptionCharacteristicPresent = checkMessageCharacteristics(message, businessDayConfig);
+            }
+
+            if (!exceptionCharacteristicPresent){
+                BusinessTime businessTime = businessDayConfig.getCurrentBusinessTime(currentDateTime);
+
+                LocalTime currentTime = currentDateTime.toLocalTime();
+                LocalTime startTime = businessTime.getStartTime();
+                LocalTime endTime = businessTime.getEndTime();
+                log.debug("start time: {}, end time: {}, current time: {}", startTime, endTime, currentTime);
 
-            if (startTime.isAfter(currentTime) || endTime.isBefore(currentTime)) {
-                ruleResponse.setSuccess(false);
-                ruleResponse.setResponseCode(NOT_BUSINESS_TIME_CODE);
-                return ruleResponse;
+                if (startTime.isAfter(currentTime) || endTime.isBefore(currentTime)) {
+                    ruleResponse.setSuccess(false);
+                    ruleResponse.setResponseCode(NOT_BUSINESS_TIME_CODE);
+                    return ruleResponse;
+                }
             }
         }
         return ruleResponse;
     }
+
+    /**
+     * Check whether the message contains characteristics which are allow to send it at not business time
+     * @param message Message to check
+     * @param businessDayConfig Configuration
+     * @return Whether the message contains characteristics which are allow to send it at not business time
+     */
+    private boolean checkMessageCharacteristics(CommunicationMessage message, BusinessDayConfig businessDayConfig) {
```

Comment:
suggest to rename `checkMessageCharacteristics` --> `isSendingAllowedByCharacteristic`

Reply:
Done